tests/framework/components/test_parser.py

```diff
@@ -75,40 +79,41 @@
 ]
 
 
-def mock_importer(path):
+def mock_importer(path: str) -> ABCMeta:
     return {
         "test_components.MockComponentA": MockComponentA,
         "test_components.MockComponentB": MockComponentB,
     }[path]
 
 
 @pytest.fixture()
-def parser(mocker) -> ComponentConfigurationParser:
+def parser(mocker: MockerFixture) -> ComponentConfigurationParser:
     parser = ComponentConfigurationParser()
-    parser.import_and_instantiate_component = mocker.Mock()
+    parser.import_and_instantiate_component = mocker.Mock()  # type: ignore [method-assign]
     return parser
 
 
 @pytest.fixture(params=[TEST_COMPONENTS_NESTED, TEST_COMPONENTS_FLAT])
-def components(request):
+def components(request: pytest.FixtureRequest) -> str:
+    assert isinstance(request.param, str)
     return request.param
 
 
 @pytest.fixture
-def import_and_instantiate_mock(mocker):
+def import_and_instantiate_mock(mocker: MockerFixture) -> Mock:
```

Comment:
Patching a MockerFixture changes it to a Mock?

Reply:
Looks like we weren't using so we deleted it and pytest still passed.